astropy/cosmology/core.py

```diff
@@ -82,6 +82,9 @@
 kB_evK = const.k_B.to(u.eV / u.K)
 
 
+_COSMOLOGY_REGISTRY = dict()
```

Comment:
Not a big deal, but by analogy with `REPRESENTATION_CLASSES` in `representations`, maybe `COSMOLOGY_CLASSES`?

Also, while I would not export it, I think it is probably appropriate as semi-public API, so I'd remove the underscore (like in `representations` and `time.formats`, and like `TIME_SCALES`, `UFUNC_HELPERS`, etc.)

Reply:
I was thinking about public scoping, but it' pretty bad if someone pops an entry. Perhaps a ``COSMOLOGY_CLASSES = MappingProxyType(_ COSMOLOGY_CLASSES)``. It's fast, subscriptable, and immutable.

Reply:
If someone really want to mess with ``_COSMOLOGY_CLASSES``, ``COSMOLOGY_CLASSES`` will mirror any changes.

Reply:
Well, it is bad too if someone does `REPRESENTATION_CLASSES.pop('cartesian')`... Or something similar with `TIME_FORMATS` (which can be imported straight from `astropy.time`). My sense would be to just make it `COSMOLOGY_CLASSES`. If the need arises, we can always subclass `dict` and protect specific keys, etc.

Reply:
I think it's more backward compatible to make an immutable object mutable, than the reverse. Methods like "pop" error on the immutable object, so starting with an immutable ``COSMOLOGY_CLASSES`` won't error if we later change it to only protecting some keys (e.g. Astropy built-in).

Also fun trick for protecting specific keys:  ``ChainMap(mutable_dict, MappingProxyType(immutable_dict))``

Reply:
I guess I still don't really see what we solve by making registries like these more indirect or hidden - the only reason to access them in the first place is to change something that is not easy to change otherwise.

But I may be wrong, so maybe best is to just stick with your initial underscore before the name to indicate this is a private attribute and users mess with it at their own risk.

Reply:
I'll push making it publicly scoped until later. The use of ``MappingProxyType`` feels equivalent to ``@property``: if one wants to mess with the private attribute, it's there, but using the public attribute can't change the underlying object.

Reply:
I like the suggestion to make ``COSMOLOGY_CLASSES`` (semi) public and it's a goal that Cosmology has 0 (or minimal) private methods.

Reply:
That's a good analogy. For now, I'd still have a very mild preference for just going with the private attribute only, but you've convinced me that the `MappingProxyType` is a good alternative. So, just do what you prefer.

Reply:
I agree with only having the private attribute.

Reply:
I think this will be my first direct commit to astropy! Now to carefully reread all the maintainer documentation before pressing that button.